Removed default dependency

Sheriff-Unit-3 opened the pull request:
This will remove the need for default to be installed for the mod to run, it will increase the support that 3d armor offers by allowing games without default to only add some aliases for crafting recipes. This will not affect existing worlds or games. all media in this pull request is licensed under CC0. Credits for sound files:

armor_glass_hit.ogg & armor_glass_break.ogg - rubberduck - licensed under CC0 - from https://opengameart.org/content/75-cc0-breaking-falling-hit-sfx

armor_metal_dig.ogg & armor_metal_break.ogg - Ogrebane - licensed under CC0 - from https://opengameart.org/content/wood-and-metal-sound-effects-volume-2

armor_wood_walk.ogg & armor_wood_dig.ogg - TinyWorlds - licensed under CC0 - https://opengameart.org/content/different-steps-on-wood-stone-leaves-gravel-and-mud

Sheriff-Unit-3 committed:
Removed default dependency

This will remove the need for default to be installed for the mod to run, it will increase the support that 3d armor offers by allowing games without default to only add some aliases for crafting recipes. This will not affect existing worlds or games.
all media in this pull request is licensed under CC0.
Credits for sound files:

armor_glass_hit.ogg & armor_glass_break.ogg - rubberduck - licensed under CC0 - from https://opengameart.org/content/75-cc0-breaking-falling-hit-sfx

armor_metal_dig.ogg & armor_metal_break.ogg - Ogrebane - licensed under CC0 - from https://opengameart.org/content/wood-and-metal-sound-effects-volume-2

armor_wood_walk.ogg & armor_wood_dig.ogg - TinyWorlds - licensed under CC0 - https://opengameart.org/content/different-steps-on-wood-stone-leaves-gravel-and-mud

SmallJoker commented on 3d_armor/api.lua:
```diff
@@ -114,10 +114,6 @@ armor = {
-		default.gui_bg..
-		default.gui_bg_img..
-		default.gui_slots..
-		default.get_hotbar_bg(0, 4.7)..
```
Unlike the other fields, this one is not a no-op. Could a helper function be introduced to generate the inventory style like I previously did for https://github.com/minetest-mods/hopper/commit/365670e906a644e6ff0da683878c997e0a52a813 (`hopper.formspec_add_list`) ? That would ease game integration as well.

Sheriff-Unit-3 replied:
So I've never used formspecs much, but from what I'm seeing in [here on line 24-28](https://github.com/luanti-org/minetest_game/blob/master/mods/default/legacy.lua) the first three are empty, not sure if that's important or not.
[Edit] I think I figured it out.

SmallJoker commented on 3d_armor_stand/init.lua:
```diff
@@ -2,10 +2,6 @@
-	default.gui_bg ..
-	default.gui_bg_img ..
-	default.gui_slots ..
-	default.get_hotbar_bg(0,3) ..
```
Make use of the aforementioned 3d_armor helper function.

SmallJoker commented on 3d_armor_stand/init.lua:
```diff
@@ -179,7 +175,11 @@ local function register_armor_stand(def)
 		groups = {choppy=2, oddly_breakable_by_hand=2},
 		is_ground_content = false,
-		sounds = default.node_sound_wood_defaults(),
+		sounds = {
```
Another helper function would be nice to have here to integrate better with other games (e.g. `hopper.metal_sounds`, link above). As a fallback, you could still use the CC0 sounds, of course.

SmallJoker reviewed (commented):
Thank you for this PR. In principle it works, and does what it says.
However, rather than removing every mention of `default`, I think that the way of helper functions is more future-proof, as explained in the comments below.

SmallJoker commented on 3d_armor_stand/sounds/armor_wood_dig.ogg:
Missing license of all new media files. Please add them to the corresponding mod's README or licensing file - whatever is available.

Sheriff-Unit-3 replied:
I'll make the above changes.
[Edit]: Everything but the formspec is done.

Sheriff-Unit-3 committed:
Added support for default sounds

Added license information to LICENSE.md in the modpack folder.

Sheriff-Unit-3 committed:
Re-added support for default's background image

SmallJoker commented:
There's a few syntax errors. Please have again a look at your code, and give it a try by testing.

Sheriff-Unit-3 committed:
Fixed mistakes from last commit

Changed out the functions used for sound to engine core functions, I did not notice a difference in-game.
Got formspecs working for games without default.

Sheriff-Unit-3 commented:
> There's a few syntax errors. Please have again a look at your code, and give it a try by testing.

New commit fixes any issues that were present before.

SmallJoker commented on shields/init.lua:
```diff
@@ -67,10 +67,10 @@ if armor.materials.wood then
 		reciprocate_damage = true,
 		on_damage = function(player, index, stack)
-			play_sound_effect(player, "default_wood_footstep")
+			armor.node_wood_sounds(player)
```
This does not play any sound.

SmallJoker commented on shields/init.lua:
```diff
@@ -94,10 +94,10 @@ if armor.materials.wood then
 		end,
 		on_destroy = function(player, index, stack)
-			play_sound_effect(player, "default_dug_metal")
+			armor.metal_break_sound(player)
```
This ignores the `shields_disable_sounds` setting which was previously checked by `play_sound_effect`. Suggestion: change `armor.metal_break_sound` (and similar) such that a SoundSpec is returned (like in `armor.node_wood_sounds`), which can then be passed as an argument to `play_sound_effect`.

Sheriff-Unit-3 replied:
>Suggestion: change `armor.metal_break_sound` (and similar) such that a SoundSpec is returned (like in `armor.node_wood_sounds`), which can then be passed as an argument to `play_sound_effect`.

I tried that first, but it would not play any sound.

Sheriff-Unit-3 replied:
> This ignores the `shields_disable_sounds` setting which was previously checked by `play_sound_effect`. 

I didn't know there was that setting, I'll add in a check for it.

Sheriff-Unit-3 replied:
The latest commit fixes the issues listed above.

Sheriff-Unit-3 committed:
Fixed Wood shield sounds and armor stand sounds.

SmallJoker commented:
Pushed as baa16e2.
I took the liberty to change the structure, but preserving your sounds and authorship by Co-authored-by.
Thank you for your contribution.
